usr/src/uts/common/fs/zfs/sa.c

```diff
@@ -1391,7 +1392,8 @@ sa_handle_get_from_db(objset_t *os, dmu_buf_t *db, void *userp,
 		error = sa_build_index(handle, SA_BONUS);
 
 		if (hdl_type == SA_HDL_SHARED) {
-			dmu_buf_init_user(&handle->sa_dbu, sa_evict, NULL);
+			dmu_buf_init_user(&handle->sa_dbu, NULL, sa_evict,
```

Comment:
Given that `sa_evict` is just a wrapper around `panic`, I think a sync callback makes more sense - it'll give a better panic stack...at the very least tying the panic to the thread kicking off the eviction.